external-providers/dotnet-external-provider/pkg/dotnet/provider.go

```diff
@@ -0,0 +1,192 @@
+package dotnet
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/swaggest/openapi-go/openapi3"
+	"github.com/go-logr/logr"
+	"github.com/konveyor/analyzer-lsp/provider"
+
+	"go.lsp.dev/jsonrpc2"
+	"go.lsp.dev/protocol"
+	"go.lsp.dev/uri"
+)
+
+type dotnetProvider struct{
+	Log logr.Logger
+}
+
+var _ provider.BaseClient = &dotnetProvider{}
+
+func NewDotnetProvider(log logr.Logger) *dotnetProvider {
+	return &dotnetProvider{
+		Log: log,
+	}
+}
+
+type stdioRWCloser struct {
+	io.Reader
+	io.Writer
+}
+
+type dotnetCondition struct {
+	Referenced referenceCondition `yaml:"referenced"`
+}
+
+// Example:
+// dotnet.referenced:
+//	namespace: System.Web.Mvc
+//	pattern: HttpNotFound
+type referenceCondition struct {
+	Namespace string `yaml:"namespace"`
+	Pattern   string `yaml:"pattern"`
+}
+
+func (r *stdioRWCloser) Close() error {
+	return nil
+}
+
+func (p *dotnetProvider) Capabilities() []provider.Capability {
+	r := openapi3.NewReflector()
+	caps := []provider.Capability{}
+	refCap, err := provider.ToProviderCap(r, p.Log, dotnetCondition{}, "referenced")
+	if err != nil {
+		p.Log.Error(err, "failed to registery capability")
+	} else {
+		caps = append(caps, refCap)
+	}
+	return caps
+}
+
+func (p *dotnetProvider) Init(ctx context.Context, log logr.Logger, config provider.InitConfig) (provider.ServiceClient, error) {
+	var mode provider.AnalysisMode = provider.AnalysisMode(config.AnalysisMode)
+	if mode != provider.FullAnalysisMode {
+		return nil, fmt.Errorf("only full analysis is supported")
+	}
+
+	// handle proxy settings
+	for k, v := range config.Proxy.ToEnvVars() {
+		os.Setenv(k, v)
+	}
+
+	codePath, err := filepath.Abs(config.Location)
+	if err != nil {
+		log.Error(err, "unable to get path to analyze")
+		panic(1)
```

Comment:
any specific reason to panic and not return the error?

Reply:
no.